Connector/Dwsim/DwsimClient.cs

```diff
@@ -47,16 +48,33 @@ public DwsimClient(
             : base(logger, config.Automation)
         {
             _logger = logger;
-            string dllPath = Path.Combine(config.Automation.DwsimInstallationPath, "DWSIM.Automation.dll");
+            var dwsimInstallationPath = config.Automation.DwsimInstallationPath;
+            if (dwsimInstallationPath == null)
+            {
+                throw new DwsimException("DWSIM installation path is not set");
+            }
+            string dllPath = Path.Combine(dwsimInstallationPath, "DWSIM.Automation.dll");
             FileVersionInfo fvi = FileVersionInfo.GetVersionInfo(dllPath);
             Version = fvi.FileVersion != null ? fvi.FileVersion : "N/A";
 
-            string propsDll = Path.Combine(config.Automation.DwsimInstallationPath, "DWSIM.FlowsheetBase.dll");
+            string propsDll = Path.Combine(dwsimInstallationPath, "DWSIM.FlowsheetBase.dll");
             Assembly assembly = Assembly.LoadFrom(propsDll);
             ResourceManager rm = new ResourceManager("DWSIM.FlowsheetBase.Properties", assembly);
-            _propMap = rm.GetResourceSet(CultureInfo.InvariantCulture, true, false)
-                .Cast<DictionaryEntry>().ToDictionary(x => x.Key.ToString(), x => x.Value.ToString());
-            _unitConverter = new UnitConverter(config.Automation.DwsimInstallationPath);
+            var resourceSet = rm.GetResourceSet(CultureInfo.InvariantCulture, true, false);
+            if (resourceSet != null)
+            {
+                foreach (DictionaryEntry entry in resourceSet)
+                {
+                    var key = entry.Key?.ToString();
+                    var value = entry.Value?.ToString();
+                    if (key != null && value != null)
```

Comment:
Thinking out loud: So previously we would have had an exception (i think?) in case a value of on the `rm` was null, now we would just ignore it, right?

Reply:
potentially yes, but this also might even never happen